### ISSUE TITLE
Re-program presentation generation flow

### ISSUE THREAD

**Issue by ronaldtse** [riboseinc/ribose-presentation-framework#1]
The current flow to rebuild CSS styling using reveal.js is cumbersome. It also requires lots of node stuff to be installed (see the README in the root and also in a presentation). When we rebuild, it also builds the CSS within the reveal.js directory causing git to think the submodule is `-dirty`.

We want to simplify this flow:
* i.e. each presentation should only contain content, and we don’t need to submodule reveal.js.

@strogonoff would you be able to do this? Thanks!

**Comment by strogonoff** [comment]
Quoting Ronald’s example:

> For example, user can have:
> - content.adoc
> Then he/she runs runs a command `gen-presentation content.adoc`, then the presentation gets saved to `content.html` that embeds (or links) reveal and necessary styling to show.

**Comment by ronaldtse** [comment]
We have created a new repository for this workflow with a specification for packaging, it is called PAW https://github.com/riboseinc/paw/ (the specification is TBD). 

i.e.

```
package-presentation content.adoc 
# => content.paw is created
```

**Comment by strogonoff** [comment]
@ronaldtse I see. I assume this means you’d want a solution that’s in some way PAW-compatible here.

Is the idea to _author_ presentation content in ADoc, and have the presentation framework tool build a PAW container with Reveal-based presentation in it?

**Comment by ronaldtse** [comment]
Yes! You're so quick 😉

**Comment by strogonoff** [comment]
Sounds good. I don’t see any interoperability issues if PAW behaves close enough to a zipped collection of HTML+assets internally. In that case we could use the same output module even in absence of PAW-supporting browser plugins, just use .zip filenames and let the user unpack.

**Comment by ronaldtse** [comment]
Yes, we will need to make some browser plugins to achieve this goal. It should basically be a zipped archive with content and metadata within.
